adagrams/game.py

```diff
@@ -1,11 +1,121 @@
+#from tests.test_wave_01 import LETTER_POOL
+import random
+import copy
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+
+SCORE_DICT = {
+    1: ['A', 'E', 'I', 'O', 'U', 'L', 'N', 'R', 'S', 'T'],
+    2: ['D', 'G'],
+    3: ['B', 'C', 'M', 'P'],
+    4: ['F', 'H', 'V', 'W', 'Y'],
+    5: ['K'],
+    8: ['J', 'X'],
+    10: ['Q', 'Z'],
+}
+
 def draw_letters():
-    pass
+    # generate list containing all letters
+    # in corresponding frequency
+    all_letters = []
+    for letter,frequency in LETTER_POOL.items():
+        for i in range(frequency):
+            all_letters.append(letter)
+    
+    hand = []
+    for i in range(10):
+        this_letter = random.choice(all_letters)
+        hand.append(this_letter)
+        all_letters.remove(this_letter)
+
+    return hand
 
 def uses_available_letters(word, letter_bank):
-    pass
+    word_bank = copy.deepcopy(letter_bank)
+    caps_word = word.upper()
+    for letter in caps_word:
+        if letter not in word_bank:
+            return False    
+        else: 
+            word_bank.remove(letter)    
+    else:
+        return True
+
```

Comment:
There's no need for the `else:` here, and many developers find the `for/else` pattern confusing (it's not common in languages other than Python). We could just do
```python
    return True
```
And it would be fine (make sure to convince yourselves why this is the case!).